.github/workflows/stable-image.yml

```diff
@@ -0,0 +1,36 @@
+name: Docker Image - Stable
```

Comment:
You do not need multiple workflows here. You can have multiple tags in one build. Normally I use latest and semver (0.0.45, etc). My normal workflow has 3 jobs, gen semver (using gitversion https://gitversion.net/docs/), build image, create github tag. Basically I match dockerImage tags with Github tags using semver. This isn't strictly required but is better practice. Having a latest tag in docker is technically frowned upon but everyone still does it. 

Example workflow of mine (https://github.com/Devcon4/kite/blob/main/.github/workflows/build.yaml)

Reply:
Yours only builds on pushes to `main` similar to the 'stable' workflow, whereas the 'latest' workflow updates each time a pull request to main is made. I could probably combine them into one though; let me check into that. 

I wanted to get the 'latest' workflow to tag the image with the git branch, but I was having issues. Using the latest for everything seems too messy. I should also use semver and tags for the stable releases as well.

Reply:
Right that is for simplicity. For small projects I don't find value in building branches other than main. I often will build main and use that as my dev env and have a second workflow that triggers off github releases for deploy/build of prod. In larger projects I will build/deploy each pull request, then merges to main deploys stage, with releases deploying prod. My prod in those cases doesn't rebuild the image, they just use the tag ref a lot of the time. Sometimes I will rebuild main in those cases because the semver would be different (no feature branch at the end). Ideally you build once and elevate rather than build many times. Semver tools like gitversion will include feature branches in the version for you. You have to be careful as well because tags (github and docker) must be URL safe where branches don't have to be, so you need to sanitize somehow (like semver tools). Another thing to note is that the ref pr actions run on is `/merge`. This is a fake ref which is the result after this pr would be merged with base. This ref includes the merge commit which would be made if this pr was merged. Other git providers don't have this ref which is annoying.

Reply:
good call, I cleared the "development action"